reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/assertj/PayloadAssert.kt

```diff
@@ -0,0 +1,266 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.integration.assertj
+
+import com.datadog.android.tests.ktx.getDouble
+import com.datadog.android.tests.ktx.getInt
+import com.datadog.android.tests.ktx.getLong
+import com.datadog.android.tests.ktx.getString
+import com.google.gson.JsonObject
+import org.assertj.core.api.AbstractObjectAssert
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.data.Offset
+import java.util.Locale
+
+internal class PayloadAssert(actual: JsonObject) :
+    AbstractObjectAssert<PayloadAssert, JsonObject>(actual, PayloadAssert::class.java) {
+
+    fun hasEnv(env: String): PayloadAssert {
+        assertThat(actual.getString(ENV_KEY)).overridingErrorMessage(
+            "Expected env to be $env but was ${actual.getString(ENV_KEY)}"
+        ).isEqualTo(env)
+        return this
+    }
+
+    fun hasSpanAtIndexWith(index: Int, block: ChildAtAssert.() -> Unit): PayloadAssert {
+        ChildAtAssert(actual, index).block()
+        return this
+    }
+
+    class ChildAtAssert(actual: JsonObject, private val index: Int) :
+        AbstractObjectAssert<ChildAtAssert, JsonObject>(actual, ChildAtAssert::class.java) {
+
+        fun hasTraceId(traceId: String): ChildAtAssert {
+            val key = TRACE_ID_KEY_FORMAT.format(Locale.US, index)
```

Comment:
> [!IMPORTANT]
> Instead of generating a key with the index for every assertions, why don't you get the object at `"span[$i]"` in the `hasSpanAtIndexWith()` function, and then pass that JsonObject as argument to this method. 
> Even though it's just tests, it is error prone to have this every where + quite inneficient.

Reply:
I was thinking to leverage those koltin extensions under the hood, also had this approach in mind. I will give it a try.

Reply:
Indeed creating all these keys with string formats is not efficient, I wanted to keep the same approach as in initial tests but maybe it's too much.